src/Tribe/Rewrite.php

```diff
@@ -534,14 +524,133 @@ public function get_canonical_url( $url, $force = false ) {
 		 */
 		$resolved = apply_filters( 'tribe_rewrite_canonical_url', $resolved, $url, $this );
 
-		if ( $found ) {
+		if ( $found_template ) {
 			// Since we're caching let's not cache unmatched rules to allow for their later, valid resolution.
 			$this->canonical_url_cache[ $url ] = $resolved;
 		}
 
 		return $resolved;
 	}
 
+	protected function resolve_link_template( $found_template, $matched_vars ) {
+		$localized_matchers = $this->get_localized_matchers();
+		$dynamic_matchers   = $this->get_dynamic_matchers( $matched_vars );
+
+		$replace = array_map( function ( $localized_matcher ) use ( $matched_vars ) {
+			if ( ! is_array( $localized_matcher ) ) {
+				// For the dates.
+				return isset( $matched_vars[ $localized_matcher ] )
+					? $matched_vars[ $localized_matcher ]
+					: '';
+			}
+
+			$query_var  = $localized_matcher['query_var'];
+			$query_vars = [ $query_var ];
+
+			if ( $query_var === 'name' ) {
+				$query_vars = array_merge( $query_vars, $this->get_post_types() );
+			}
+
+			if ( ! array_intersect( array_keys( $matched_vars ), $query_vars ) ) {
+				return '';
+			}
+
+			/*
+			 * We use `end` as, by default, the localized version of the slug in the current language will be at the
+			 * end of the array.
+			 * @todo here we should keep a map, that has to generated at permalink flush time, to map locales/slugs.
```

Comment:
:warning: Comment refers to a TODO task "here we should keep a map, that has to generated at permalink flush time, to map locales/slugs"

<sub>hash: f3fa0b5a1459de790f74a0186dfea822</sub>